lib/optimize.js

```diff
@@ -0,0 +1,156 @@
+                return false;
+            }
+
+            if (!_isReducerCall(expr.right)) {
+                optimization_info.invalid = 'reduce other than sum(field)';
+                return false;
+            }
+
+            let reducer = _getReducerName(expr.right);
+
+            if (reducer !== 'sum') {
+                optimization_info.invalid = 'reduce other than sum(field)';
+                return false;
+            }
+
+            let arg = expr.right.arguments[0];
+            if (arg.type !== 'StringLiteral') {
+                optimization_info.invalid = 'reduce other than sum(field)';
+                return false;
+            }
+
+            metrics.push(arg.value);
+            defaults.push(reducerDefaultValue(reducer));
+        }
+
+        if (graph.node_has_option(reduce, 'every')) {
+            let every = graph.node_get_option(reduce, 'every');
+
+            if (every.eq(JuttleMoment.duration(1, 'm'))) {
+                dimensions.push('dateHour', 'minute');
+
+            } else if (every.eq(JuttleMoment.duration(1, 'h'))) {
+                dimensions.push('dateHour');
+
+            } else if (every.eq(JuttleMoment.duration(1, 'd'))) {
+                dimensions.push('date');
+
+            } else if (every.eq(JuttleMoment.duration(1, 'w'))) {
+                dimensions.push('year', 'week');
+
+            } else if (every.eq(JuttleMoment.duration(1, 'M'))) {
+                dimensions.push('year', 'month');
+
+            } else if (every.eq(JuttleMoment.duration(1, 'Y'))) {
+                dimensions.push('year');
+
+            } else {
+                optimization_info.invalid = 'reduce with interval other than 1 minute, hour, day, week, month or year';
+                return false;
+            }
+        }
+
+        let groupby = graph.node_get_option(reduce, 'groupby');
+        let grouped = groupby && groupby.length > 0;
+        if (grouped) {
+            let BAD_GROUPBY = ['time', 'date', 'dateHour', 'year', 'month', 'minute'];
+            let META_GROUPBY = ['viewId', 'view', 'webPropertyId', 'webProperty'];
+            let bad = _.intersection(groupby, BAD_GROUPBY);
+            if (bad.length !== 0) {
+                optimization_info.invalid = 'reduce group by ' + bad[0];
+                return false;
+            }
+
+            // If a viewId is specified in the read options, then it's invalid
+            // to group by any of the meta dimensions.
+            if (graph.node_get_option(read, 'viewId')) {
+                bad = _.intersection(groupby, META_GROUPBY);
+                if (bad.length !== 0) {
+                    optimization_info.invalid = '-viewId with reduce group by ' + bad[0];
+                    return false;
+                }
+            }
+
+            dimensions = dimensions.concat(_.difference(groupby, META_GROUPBY));
+            if (_.contains(groupby, 'view') || _.contains(groupby, 'viewId')) {
```

Comment:
Not exactly... the `metagroup` ends up affecting a couple things in the query:
1. If the metagroup is `view`, then all views are included in the results, not just the "All Web Site Data" view.
2. If the metagroup is set at all, then we tack on the full view metadata (including view, viewId, webProperty, and webPropertyId) into each point.

In either case, the specific groupby keys that the user selected are then handled by the native juttle `reduce` that's left in the flowgraph since optimization always returns false.